### ISSUE TITLE
Store contig names in a `contig_id` array

### ISSUE THREAD

**Issue by timothymillar** [pystatgen/sgkit#1035]
Currently these are stored in a dataset attribute, but it would be more consistent to store them as an array of dimension "contigs".

Storing contig names as an attribute can get messy when using a fragmented genome with thousands of scaffolds because, by default, the dataset attrs are printed in the repl representation.

Furthermore, storing them as an array would simplify some operations. E.g., in the GWAS tutorial

```python
ds["variant_contig_name"] = (["variants"], [ds.contigs[contig] for contig in ds["variant_contig"].values])
```

Could be simplified to

```python
ds["variant_contig_name"] = ds.contig_id[ds.variant_contig]
```

But these are minor issues, so I understand if others would rather avoid a breaking change.

**Comment by tomwhite** [comment]
I support this change, but we should think about compatibility since it affects the storage format. (See https://github.com/pystatgen/vcf-zarr-spec/blob/main/vcf_zarr_spec.md)

I think we could have code that looks for the variable then falls back to the attribute if it is not present (as it wouldn't be in older datasets). Alternatively we could document how users can manually migrate from the old version to the new version by showing a snippet of code that creates the variable from the attribute.

BTW there is also a `filters` attribute that is created by the VCF reader that should be changed similarly.

**Comment by tomwhite** [comment]
I created a PR to update the spec [here](https://github.com/pystatgen/vcf-zarr-spec/pull/5), and started on a draft implementation [here](https://github.com/pystatgen/sgkit/pull/1054).

I'm interested to hear what folks think about this change. Is it something we should phase in gradually (e.g. add the new arrays, and duplicate information for a while), or switch over in one release? I've included a way for sgkit itself to fall back to accessing contigs via an attribute if present, but that won't work for user code that explicitly accesses the attribute once it has been removed.
